workload/aspnetapp-canary.yaml

```diff
@@ -0,0 +1,133 @@
+#  ------------------------------------------------------------
+#   Copyright (c) Microsoft Corporation.  All rights reserved.
+#   Licensed under the MIT License (MIT). See License.txt in the repo root #  for license information.
+#  ------------------------------------------------------------
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: aspnetapp-canary-deployment
+  namespace: a0042
+  labels:
+    app.kubernetes.io/name: aspnetapp-canary
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: aspnetapp-canary
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: aspnetapp-canary
+    spec:
+      automountServiceAccountToken: false
+      securityContext:
+        runAsUser: 10001
+        runAsGroup: 3000
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app.kubernetes.io/name
+                operator: In
+                values:
+                - aspnetapp
+            topologyKey: "kubernetes.io/hostname"
+        podAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app.kubernetes.io/name
+                  operator: In
+                  values:
+                  - traefik-ingress-ilb
+              topologyKey: "kubernetes.io/hostname"
+      containers:
+      - name: aspnet-webapp-sample
+        image: mcr.microsoft.com/dotnet/samples:aspnetapp
+        imagePullPolicy: Always
+        securityContext:
+          allowPrivilegeEscalation: false
+          readOnlyRootFilesystem: true
```

Comment:
I was reading that this is literally unsupported for aspnet images.  Even with the COMPlus_EnableDiagnotics flag set. ASP.NET will page to filesystem if it needs to.  Sounds like it could be dangerous to block this from happening.  I know we are not here to show "the best possible aspnetcore" container spec ... but just thought I would mention it.  I caught it in regulated and remove this (or set it to false, I forget which)

Reply:
Also, I know completely unrealted to what your change is.  Take/leave/backlog -- as  you wish.

Reply:
yeah, the thing is that we got reports with some warnings at least when configuring it the other way around.

changed | Addressed [69318](https://github.com/mspnp/aks-secure-baseline/pull/185/commits/693185325032f8e433a591bff08a3d447d79b2d0)